agent/engine/engine_sudo_linux_integ_test.go

```diff
@@ -238,6 +262,31 @@ func TestFirelensFluentbit(t *testing.T) {
 	assert.Error(t, err)
 }
 
+// getPortFromAddr returns the port part of an address in format "http://<addr>:<port>".
+func getPortFromAddr(addr string, t *testing.T) string {
+	parts := strings.Split(addr, ":")
```

Comment:
nit: I think `net.SplitHostPort(u.Host)` might be more appropriate here, although you need an extra step: 
https://play.golang.org/p/glnDMk9BkPr